src/BinDeps.jl

```diff
@@ -561,6 +561,20 @@ function eval_anon_module(context, file)
     return
 end
 
+"""
+    glibc_version()
+
+For Linux-based systems, return the version of glibc in use. For non-glibc Linux and
+other platforms, returns `nothing`.
+"""
+function glibc_version()
+    Compat.Sys.islinux() || return
+    ptr = Libdl.dlsym_e(Libdl.dlopen_e("libc"), :gnu_get_libc_version)
```

Comment:
`""` should be enough and will avoid dealing with sonames.
Also, you should either use `Libdl.dlopen`, or use `Libdl.dlopen_e` with check on return value before passing it to `dlsym`.

Reply:
What do you mean "`""` should be enough"? Good call regarding checking the return value of `dlopen(_e)`.

Reply:
Replace the only string in this line with `""`.

Reply:
I'm confused, that doesn't seem to do the same thing:
```
julia> Libdl.dlopen_e("")
Ptr{Void} @0x0000000000000000

julia> Libdl.dlopen("libc")
Ptr{Void} @0x0000000112fafc90
```

Reply:
This behavior might be version dependent so the more reliable one would be to use `ccall(:jl_dlopen, Ptr{Void}, (Ptr{Void}, UInt32), C_NULL, 0)`.

Reply:
Which behavior is version-dependent? `dlopen_e("")` returns a null pointer and `dlopen_e("libc")` seems to work as expected on all of 0.3-0.7.

Reply:
Version dependent as in libc version dependent. Not julia version dependent. `libc` is not guarantee to exist.

Reply:
Oh! Weird, I didn't realize that libc could be absent altogether.

Reply:
`libc.so` does not exist on most systems.

Reply:
And even when it does, it's basically never the one you want.

Reply:
> on most systems

On most Linux systems? This code only gets run on Linux since the other platforms return early.

Reply:
Correct. It's a linker script for all glibc.